### DISCUSSION TITLE
How to multi-thread Windows.Storage.StorageFile file access in WinUI

### DISCUSSION THREAD

**Discussion by HEIC-to-JPEG-Dev** [microsoft/CsWinRT#1412] [General]
I've tried to multi-thread read files using StorageFile (an array of 100 files). But this crashes WinUI with an obscure error. Run the same code sequentualy and it works, just slowly.

In my scenario I actually have about 100,000 files that I need some basic information from. These files are placeholder files and indexed so I need to be careful how I look at them (I don't want them to be hydrated). StorageFile can get the information and I can use the file system indexer for precaching, but going through them one at a time is too time consuming.

**Comment by manodasanW** [comment]
@HEIC-to-JPEG-Dev Can you provide more details on the error code / message that you are seeing it crash with.

**Reply by HEIC-to-JPEG-Dev** [reply]
Actually this can be closed. I got passed the error but storage doesn’t support multi-threaded access. The runtime broker brings it all down to a single thread so multi-threading becomes moot

**Comment by HEIC-to-JPEG-Dev** [comment]
Pretty sure csWinRT is dead - no responses from Microsoft on issues & bugs raised last year, no roadmap. I think all efforts are going into WPF now ?

**Reply by manodasanW** [reply]
Our focus has been on getting our .NET AOT changes for CsWinRT stabilized for prerelease as it will bring improvements that hopefully brings us closer to UWP performance and will also improve performance for non AOT scenarios too.  At least with respect to this discussion post, was there anything left as I assumed you figured out what was happening based on your last comment indicating runtimebroker.

**Reply by HEIC-to-JPEG-Dev** [reply]
Thank you for an updates. Any response, even just "we're aware" is so appreciated. 
I went through a profiling excercise and two scenarios, one complex and one simple. In both cases, running debug and release on winUI and UWP, WinUI took a minimum of 10x longer, sometime much more. I'm hoping the AOT stuff helps as I'm going to start my new project in WinUI with that expectation.
Thank you.
